src/frames.rs

```diff
@@ -42,24 +42,25 @@ impl Default for UnresolvedFrames {
 
 impl Clone for UnresolvedFrames {
     fn clone(&self) -> Self {
-        let slice = self.slice().clone();
+        let slice = self.inited_slice().clone();
         Self::new(slice.frames, slice.thread_name, slice.thread_id)
     }
 }
 
 impl Debug for UnresolvedFrames {
     fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
-        self.slice().fmt(f)
+        self.inited_slice().fmt(f)
     }
 }
 
 impl UnresolvedFrames {
-    #[allow(clippy::uninit_assumed_init)]
     pub fn new(bt: &[Frame], tn: &[u8], thread_id: u64) -> Self {
         let depth = bt.len();
-        let mut frames: [Frame; MAX_DEPTH] =
+        let mut frames: [MaybeUninit<Frame>; MAX_DEPTH] =
```

Comment:
Can you verify the cost of creating a default `Frame`? If it's cheap enough, then no need to use complicated syntax.

Reply:
The problem is that the `Frame` didn't implement `Default`.

Reply:
This PR replaces the original `std::mem::MaybeUninit::uninit().assume_init()` (which is not preferred) into multiple steps:

Run ``std::mem::MaybeUninit::uninit()` and init them, and then `assume_init()` (actually [`slice_assume_init`](https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.slice_assume_init_ref), but I didn't call this function directly considering it cannot be used in stable compiler.)

Reply:
How about using `SmallVec`? So no allocation is needed and memory is maintained properly. smallvec is the indirect dependency of the crate already.